lib/src/ovis_util/util.c

```diff
@@ -990,21 +990,53 @@ const char* ovis_errno_abbvr(int e)
 	return "UNKNOWN_ERRNO";
 }
 
+/** should be bigger than any expected strerror string.
+ * In 2020 ~50 was the upper limit.
+ */
+#define OVIS_SYS_ERRLEN 80
+
+/** should be bigger than any expected errno value.
+ * In 2020 ~140 was the upper limit.
+ */
+#define OVIS_SYS_NERR 256
+
+static char ovis_sys_errlist[OVIS_SYS_NERR][OVIS_SYS_ERRLEN];
+static int ovis_sys_errlist_initialized;
+static int ovis_sys_nerr;
+/* glibc has dropped support for sys_errlist.
+ * we copy it via strerror on first call.
+ * Make an unused call to ovis_strerror from main to avoid races.
+ */
+static void ovis_sys_errlist_init() {
+	int i = 0;
+	char unknown[OVIS_SYS_ERRLEN];
+	strerror_r(OVIS_SYS_NERR, unknown, sizeof(unknown));
+	while (!isdigit(unknown[i]) && unknown[i] != '\0')
```

Comment:
Why do we want to strip off the rest of the string beginning at any digit? Needs a code comment if this is the right thing to do.

Reply:
The current gnu libc uses a format for unknown errno values of "Unknown error %d". This code assumes that format will persist and allow us to identify the actual value of sys_nerr (the first occurance of the common prefix, whatever it is).
If the assumption fails, we fall back to OVIS_SYS_NERR -1 and everything will still run 'correctly' but with different string formatting for the 'unknown' error cases.